qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/VisitTypeSteps.java

```diff
@@ -0,0 +1,119 @@
+    private VisitTypePage visitTypePage;
+    private VisitTypeListPage visitTypeListPage;
+    private String visitTypeUuid;
+    private final String VISIT_TYPE_NAME = RandomStringUtils.randomAlphabetic(5);
+    private static final String VISIT_TYPE_DESCRIPTION = "Description for Visit Type test workflow";
+
+    @Before(RunTest.HOOK.SELENIUM_VISIT_TYPE)
+    public void setup() {
+        initiateWithLogin();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_VISIT_TYPE)
+    public void teardown() {
+        RestClient.delete("visittype/" + visitTypeUuid, true);
+        quit();
+    }
+
+    @Given("a user clicks on the system administration link from the home page")
+    public void launchSystemAdministrationPage() {
+        administrationPage = homePage.goToAdministration();
+    }
+
+    @Then("the system loads the administration page")
+    public void systemLoadsAdministrationPage() {
+        assertTrue(textExists("Manage Visit Types"));
+    }
+
+    @And("a user clicks on the manage Visit Types link")
+    public void launchAdvancedAdministrationPage() {
+        visitTypeListPage = administrationPage.goToVisitTypePage();
+    }
+
+    @When("a user clicks on the add visit type link on the manage visit type page")
+    public void clickOnAddVisitTypeLink() {
+        visitTypePage = visitTypeListPage.addVisitType();
+    }
+
+    @Then("the system loads the visit type form page")
+    public void systemLoadsVisitTypeFormPage() {
+        assertTrue(textExists("Description"));
+    }
+
+    @And("a user fills the add visit type form")
+    public void fillAddVisitTypeForm() {
+        visitTypePage.setName(VISIT_TYPE_NAME);
+        visitTypePage.setDescription(VISIT_TYPE_DESCRIPTION);
+    }
+
+    @And ("a user saves visit type")
+    public void saveVisitType() {
+        visitTypePage.save();
+    }
+
+    @Then("a visit type is saved successfully")
+    public void systemAddsVisitType() {
+        assertTrue(textExists("Visit Type saved"));
+    }
+
+    @When("a user clicks on the already saved visit type link")
+    public void clickAlreadySavedVisitType() {
+        visitTypePage = visitTypeListPage.clickAlreadySavedVisitType();
+    }
+
+    @And("a user edits the visit type name")
+    public void editVisitType() {
+        visitTypePage.setName("edited VisitTypeName");
+        visitTypePage.save();
+    }
+
+    @And ("a user fills in the reason for retiring visit type")
+    public void setRetireReason() {
+        visitTypePage.setRetireReason("retire reason");
+    }
+
+    @And("a user clicks on retire visit type button")
+    public void retireVisitType() {
+        visitTypePage.retire();
+    }
+
+    @And("a user clicks on delete visit type forever button")
+    public void deleteVisitType() {
+        visitTypePage.delete();
+    }
+
+    @Then("a visit type is deleted successfully")
+    public void systemDeletesAVisitType() {
+        assertTrue(textExists("Visit Type deleted forever successfully"));
+    }
+}
```

Comment:
Add an empty line at the EOF to cater for git collaborators.

Reply:
@Muta-Jonathan did you take a look at the comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/317/files#r788430289)?